uw-frame-components/portal/notifications/controllers.js

```diff
@@ -6,31 +6,64 @@ define(['angular'], function(angular) {
 
   app.controller('PortalNotificationController', [ '$scope',
                                                    '$rootScope',
+                                                   '$location',
                                                    'NOTIFICATION',
                                                    'SERVICE_LOC',
                                                    'filterFilter',
                                                    'notificationsService',
                                            function($scope,
                                                     $rootScope,
+                                                    $location,
                                                     NOTIFICATION,
                                                     SERVICE_LOC,
                                                     filterFilter,
                                                     notificationsService){
-                                                      
-    
+
+
+    var configurePriorityNotificationScope = function(data){
+      $scope.priorityNotifications = filterFilter(data, {priority : true});
+      if ($scope.priorityNotifications && $scope.priorityNotifications.length > 0) {
+        $scope.hasPriorityNotifications = true;
+        $('.page-content').addClass('has-priority-nots');
+        if($scope.headerCtrl) {
+          $scope.headerCtrl.hasPriorityNotifications = true;
+        }
+      } else {
+        $('.page-content').removeClass('has-priority-nots');
+        $scope.hasPriorityNotifications = false;
+        if($scope.headerCtrl) {
+          $scope.headerCtrl.hasPriorityNotifications = false;
+        }
+      }
+    };
+
+    var clearPriorityNotificationFlags = function() {
+      $scope.priorityNotifications = [];
+      $scope.hasPriorityNotifications = false;
+      if($scope.headerCtrl) {
+        $scope.headerCtrl.hasPriorityNotifications = false;
+      }
+      $('.page-content').removeClass('has-priority-nots');
+    }
+
+    $scope.$on('$locationChangeStart', function(event) {
+      if ($location.url().indexOf('notification') === -1) {
```

Comment:
Does this mean that when viewing content with `fname` `some-notification`, the priority notification header would be suppressed? How about if we invented a `/settings/notifications`?